src/rendering/dashboard.cpp

```diff
@@ -137,8 +149,18 @@ void Dashboard::render(glm::vec2& penPosition) {
         return;
     }
 
+    auto now = std::chrono::high_resolution_clock::now();
+    // count returns values in nanoseconds, _refreshRate is in milliseconds
+    const bool needsUpdate = (now - _lastRefresh).count() > _refreshRate * 1e6;
```

Comment:
I don't believe there's any guarantee about the period unit is for `high_resolution_clock` (or `steady_clock`). If you want to be sure, you'd rather use `duration_cast` or cast the `_refreshRate` variable:

```suggestion
    const bool needsUpdate = (now - _lastRefresh) > std::chrono::milliseconds(_refreshRate);
```

(untested)